### ISSUE TITLE
Add bounded/* to the schema doc [phab:D1116]

### ISSUE THREAD

**Issue by kythe** [google/kythe#1472]
_Created by **salguarnieri** at 2016-10-24 23:54:02_: 

Add documentation of the bounded/upper and bounded/lower edges to the
schema doc. These are used when a type variable is bounded by other
types. In java, this is done with <? extends String> or <? super
String>.

Also, add a test for for Java generics that uses <? super String> since
all the other bounded tests were all covariant.

Also, add IJwB project setup for java indexer so it is easier for people
to set up and use IntelliJ to hack on the Java indexer.

**Comment by kythe** [comment]
_**salguarnieri** wrote at 2016-10-24 23:57:26_:

  - fix typo

**Comment by kythe** [comment]
_**zarko** commented inline (context lost during migration) at 2016-10-26 00:21:25_:

I would say:

Kythe leaves the interpretation of unbounded <<absvar>> nodes to each language. For example, an <<absvar>> with no *bounded* edges in Java may be assigned any subtype of `Object`, but no primitive type.

("most generic" is difficult to interpret, and we can't say "top type" because Java's primitive types are disjoint from its reference types.)

**Comment by kythe** [comment]
_**zarko** commented inline (context lost during migration) at 2016-10-26 00:21:26_:

I would say:

`<<absvar>> A is *bounded/upper* by B when A is constrained to be a subtype of B. A is *bounded/lower* by B when A is constrained to be a supertype of B.`

"contravariant subtype" could be confusing and should at least be replaced with "supertype" (with "covariant" dropped above).

**Comment by kythe** [comment]
_**zarko** commented inline (context lost during migration) at 2016-10-26 00:21:26_:

This appears as a separate code-formatted paragraph; it should be merged with the `Notes::` section above.

**Comment by kythe** [comment]
_**salguarnieri** wrote at 2016-10-26 01:14:37_:

  - Clean up phrasing
  - spell check

**Comment by kythe** [comment]
_**salguarnieri** commented inline (context lost during migration) at 2016-10-26 16:53:27_:

Changed to a list

**Comment by kythe** [comment]
_**zarko** commented inline (context lost during migration) at 2016-10-26 19:41:31_:

This ends up formatted as a numeric list, e.g.

```
[absvar] A is bounded/upper by B...
  1. [absvar] A is bounded/lower by B...
```

A quick and unsatisfying fix is to move `of` down to the next line, so it begins `  of B. <<absvar>>...`

**Comment by kythe** [comment]
_**salguarnieri** wrote at 2016-10-26 19:56:32_:

  - Line formatting

**Comment by kythe** [comment]
_**zarko** accepted this revision at 2016-10-27 00:06:31_

**Comment by kythe** [comment]
_**salguarnieri** closed via commit 0d1f937696bc3443e22bdfa888931c092a392ed1 at 2016-10-27 00:08:48_
